repository.go

```diff
@@ -46,47 +47,54 @@ func (r *Repository) CloneInto(path string, bare bool) error {
 	})
 
 	if err == git.ErrRepositoryAlreadyExists {
-		// Pull instead of clone
-		if gitRepo, err = git.PlainOpen(path); err == nil {
-			// we need to check whether it's a bare repo or not.
-			// if not we should pull, if it is then pull won't work
-			if isBare, bErr := isBare(gitRepo); bErr == nil && !isBare {
-				if w, wErr := gitRepo.Worktree(); wErr != nil {
-					err = wErr
-				} else {
-					err = w.Pull(&git.PullOptions{
-						Auth:     auth,
-						Progress: os.Stdout,
-					})
-				}
-			}
+		// as the repo already exists, we just plain open it
+		gitRepo, err = git.PlainOpen(path)
+		if err != nil {
+			return err
 		}
 	}
 
-	switch err {
-	case transport.ErrEmptyRemoteRepository:
-		log.Printf("%s is an empty repository", r.FullName)
-		//  Empty repo does not need backup
-		return nil
-	default:
+	var remotes []*git.Remote // we only create this variable here so we don't have to use := with the Remotes() call, which would create a new err
+	remotes, err = gitRepo.Remotes()
+	if err != nil {
 		return err
-	case git.NoErrAlreadyUpToDate:
-		log.Printf("No need to pull, %s is already up-to-date", r.FullName)
-		// Already up to date on current branch, still need to refresh other branches
-		fallthrough
-	case nil:
-		// No errors, continue
-		err = gitRepo.Fetch(&git.FetchOptions{
+	}
+
+	// we iterate over all the remotes, and we fetch all their heads
+	for _, remote := range remotes {
+		err = remote.Fetch(&git.FetchOptions{
 			Auth:     auth,
 			Progress: os.Stdout,
 			Tags:     git.AllTags,
 			Force:    true,
+			RefSpecs: []config.RefSpec{"+refs/heads/*:refs/heads/*"},
```

Comment:
This is basically the magic to automatically update all local branches during the fetch operation.

Reply:
Yeah this is a good change on its own already because I thought fetch would already fetch all refs, not just the local ones.